src/HomeworkDataTypes.java

```diff
@@ -0,0 +1,431 @@
+        int counter = 0;
+        for (int i = a; i <= b; i++) {
+            sum += i;
+            counter++;
+        }
+        System.out.println("Sum for range " + a + " to " + b + " : " + sum);
+        avg = sum / counter;
+        System.out.println("Average for range " + a + " to " + b + " : " + avg);
+    }
+
+    /* -------------------------------------------------------------- */
+    // 3. Write a program that will print the first x fibonacci numbers. Also calculate the average value. For example
+    // For an input of x = 20 the output should be:
+    // The first 20 Fibonacci numbers are:
+    // 1 1 2 3 5 8 13 21 34 55 89 144 233 377 610 987 1597 2584 4181 6765
+    // The average is 885.5
+
+    public void printFibonacciNumbers(int a) {
+        int myArray[] = new int[a];
+        int i;
+        int sum = 0;
+        myArray[0] = 1;
+        myArray[1] = 1;
+        System.out.println("The first 20 Fibonacci numbers are: ");
+        System.out.println(myArray[0]);
+        System.out.println(myArray[1]);
+        for (i = 2; i <= myArray.length - 1; i++) {
+            myArray[i] = myArray[i - 1] + myArray[i - 2];
+            sum += myArray[i];
+            System.out.println(myArray[i] + " ");
+        }
+        double avg = (double) (sum + 2) / a;
+        System.out.println("The average is: " + avg);
+    }
+
+    /* -------------------------------------------------------------- */
+    // 4, Write a java program that will test whether the first and the last element of an int array are the same.
+    // We can assume the array has 2 or more elements.
+    // For example:
+    // For the following input [13,6,2,8,2,3] the output should be : FALSE
+    // For the following input [13,6,2,8,2,13] the output should be : TRUE
+
+    public void testSame(int myArray[]) {
+        boolean isSame = false;
+        if (myArray[0] == myArray[myArray.length - 1]) {
+            isSame = true;
+            System.out.println("TRUE");
+        } else {
+            isSame = false;
+            System.out.println("FALSE");
+        }
+    }
+
+    /* -------------------------------------------------------------- */
+    // 5. Write a program that test whether 2 int arrays are equal.
+    // Equality in this case means that both arrays have the same number of elements an element
+    // on a specific index in first array is equal to the element on the same index in the second array.
+    // For example:
+    // For the following arrays [1,2,3,4] and [1,2,3,4] the output should be TRUE
+    // For the following arrays [] and [] the output should be TRUE
+    // For the following arrays [1,3,2] and [1,3] the output should be FALSE
+    // For the following arrays [1,2,3,4] and [1,2,4,3] the output should be FALSE
+    public void arrayEquity(int first[], int second[]) {
+        int sizeFirst = first.length;
+        int sizeSecond = second.length;
+        int counterForEquity = 0;
+        if (sizeFirst == 0 && sizeSecond == 0) {
```

Comment:
Condifitia asta nu mai este enaparat necesara, este de ajuns sa verifici daca sunt diferite si sa returnezi false, si daca trece de acest if inseamna ca au lungimile egale. Si poti direct sa iterezi, dar lungimea este 0 for-ul nu o sa faca nimic